packages/datatrak-web/src/utils/useUserCountries.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+import { useState } from 'react';
+import { useEntities, useUser } from '../api/queries';
+import { Entity } from '../types';
+
+export const useUserCountries = () => {
```

Comment:
Great to see all these edge cases thought out and handled. I think it would be nice to re-organise where the code lives a bit to make it easier to maintain. I think that it could be nice to create a CountrySelect component that could live in `features` that has this and the JSX related to it either in the same file or a folder. I'm open to other organisation too, I'm just keen to move this code out of utils since it is pretty specific to one bit of ui.